Libraries/src/Amazon.Lambda.Annotations.SourceGenerator/Validation/RouteParametersValidator.cs

```diff
@@ -0,0 +1,56 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Amazon.Lambda.Annotations.SourceGenerator.Models;
+using Amazon.Lambda.Annotations.SourceGenerator.Models.Attributes;
+
+namespace Amazon.Lambda.Annotations.SourceGenerator.Validation
+{
+    public static class RouteParametersValidator
+    {
+        private static readonly HashSet<string> NonFromRouteAttributes = new HashSet<string>
+        {
+            TypeFullNames.FromHeaderAttribute,
+            TypeFullNames.FromQueryAttribute,
+            TypeFullNames.FromBodyAttribute,
+            TypeFullNames.FromServiceAttribute
+        };
+
+        public static (bool isValid, IList<string> missingParameters) Validate(HashSet<string> routeParameters, IList<ParameterModel> lambdaMethodParameters)
+        {
+            var missingRouteParams = new List<string>();
+            var isValid = true;
+
+            foreach (var routeParam in routeParameters)
+            {
+                var fromRouteMethodParam = lambdaMethodParameters
+                    .FirstOrDefault(p => p.Attributes.Any(att => (att as AttributeModel<FromRouteAttribute>)?.Data.Name == routeParam));
+
+                // There exists a parameter attributed with FromRouteAttribute
+                if (fromRouteMethodParam != null)
+                {
+                    continue;
+                }
+
+                var nameMatchingMethodParam = lambdaMethodParameters.FirstOrDefault(p => p.Name == routeParam);
+
+                // There doesn't exist a parameter with matching name as template parameter
```

Comment:
Same comment as above that is is a partial thought comment.

Reply:
Reworded.